secboot/secboot_tpm.go

```diff
@@ -269,6 +270,28 @@ func UnlockVolumeIfEncrypted(disk disks.Disk, name string, encryptionKeyDir stri
 	return filepath.Join("/dev/disk/by-partuuid", partUUID), false, nil
 }
 
+// UnlockVolumeUsingSealedKeyIfEncrypted unlocks existing volume using the using
+// the provided key. The path to the device node is returned.
+func UnlockEncryptedVolumeUsingKey(disk disks.Disk, name string, key []byte) (string, error) {
```

Comment:
this doesn't seem tested, also not sure it should live in the _tpm file in the end, but likely ok for now.

Reply:
Added some tests now.